src/stream/src/executor/sort_buffer.rs

```diff
@@ -11,385 +11,96 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-#![cfg_attr(not(test), expect(dead_code))]
 
-use std::collections::BTreeMap;
+use std::marker::PhantomData;
 use std::ops::Bound;
 
-use futures::stream::select_all;
-use futures::{stream, StreamExt, TryStreamExt};
-use futures_async_stream::for_await;
-use risingwave_common::array::{DataChunk, Op, StreamChunk};
-use risingwave_common::catalog::Schema;
+use futures::stream;
+use futures_async_stream::try_stream;
+use risingwave_common::array::stream_record::Record;
 use risingwave_common::hash::VnodeBitmapExt;
-use risingwave_common::row::{self, AscentOwnedRow, OwnedRow, Row, RowExt};
-use risingwave_common::types::{ScalarImpl, ToOwnedDatum};
-use risingwave_common::util::chunk_coalesce::DataChunkBuilder;
+use risingwave_common::row::{self, OwnedRow, Row};
+use risingwave_common::types::ScalarImpl;
 use risingwave_storage::store::PrefetchOptions;
 use risingwave_storage::StateStore;
 
-use super::{Barrier, PkIndices, StreamExecutorResult};
+use super::StreamExecutorError;
 use crate::common::table::state_table::StateTable;
 
-/// [`SortBufferKey`] contains a record's timestamp and pk.
-type SortBufferKey = (ScalarImpl, AscentOwnedRow);
-
-/// [`SortBufferValue`] contains a record's value and a flag indicating whether the record has been
-/// persisted to storage.
-/// NOTE: There is an exhausting trade-off for which structure to use for the in-memory buffer. For
-/// example, up to 8x memory can be used with [`OwnedRow`] compared to the `CompactRow`. However, if
-/// there are only a few rows that will be temporarily stored in the buffer during an epoch,
-/// [`OwnedRow`] will be more efficient instead due to no ser/de needed. So here we could do further
-/// optimizations.
-type SortBufferValue = (OwnedRow, bool);
-
-/// [`SortBuffer`] is a common component that consume an unordered stream and produce an ordered
-/// stream by watermark. This component maintains a state table internally, which schema is same as
-/// its input and output. Generally, the component acts as a buffer that output the data it received
-/// with a delay.
 pub struct SortBuffer<S: StateStore> {
```

Comment:
We will use the sort buffer without any cache at first to implement executors so that we can do not care the modification on the sort buffer. and add the cache later.